meter/publisher.cc

```diff
@@ -85,17 +86,26 @@ rapidjson::Document MeasurementsToJson(
   int64_t added = 0;
   Document payload{kObjectType};
   auto& alloc = payload.GetAllocator();
+  auto logger = Logger();
 
   Value common_tags_section{kObjectType};
 
   Value json_metrics{kArrayType};
+  auto should_debug = logger->should_log(spdlog::level::debug);
   for (auto it = first; it < last; it++) {
     const auto& measure = *it;
     if (std::isnan(measure->value())) {
+      if (should_debug) {
```

Comment:
Will the logger not check this internally? Is the reason for the guard because `measure->all_tags()` is potentially expensive?